### ISSUE TITLE
RHS of dead assign is removed, leading to invalid behavior

### ISSUE THREAD

**Issue by jaskarth** [Vineflower/vineflower#274]
Quiltflower:
```java
public void addHandler(final HandleWebCommandSupport handler) {
      synchronized (this) {
         ;
      }
   }
```
Fernflower:
```java
public void addHandler(final HandleWebCommandSupport handler) {
        synchronized(this){}

        try {
            ArrayBuffer var10000 = (ArrayBuffer)this.handlers.$plus$eq(handler);
        } catch (Throwable var4) {
            throw var4;
        }

    }
```
[DefaultWebCommands.zip](https://github.com/QuiltMC/quiltflower/files/11362794/DefaultWebCommands.zip) Apache 2.

**Comment by Kroppeb** [comment]
Do you have an source code example that produces this behavior?

**Comment by jaskarth** [comment]
It's in scala, but the source of the above classfile: <https://github.com/playframework/playframework/blob/main/core/play/src/main/scala/play/core/system/WebCommands.scala#L40>

**Comment by jaskarth** [comment]
Yes, but they're wrong for different reasons. Fernflower can't see the synchronized block there and Quiltflower destroys the content of the block

**Comment by jaskarth** [comment]
This bug seems to be fixed in 1.10. Bisecting points to 62cfd6b fixing it, as a side effect.
